[Security Solution][Attacks/Alerts][Attack flyout] Format table fields and values

agusruidiazgd opened the pull request:
## Summary

Closes: https://github.com/elastic/kibana/issues/243649

This PR updates the Table tab formatting inside the Attack Details Flyout to match the styling and layout used in others flyouts in Security Solution.
The goal is to ensure visual and structural consistency across the UI.


https://github.com/user-attachments/assets/28a63b5c-a621-4bb3-8a55-445e78ad5aa8

agusruidiazgd committed:
[Security Solution][Attacks/Alerts][Attack flyout] Format table fields and values

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)

PhilippeOberti reviewed (commented):
Haven't looked at the code yet, but could you add `/x-pack/solutions/security/plugins/security_solution/public/flyout/attack_details @elastic/security-threat-hunting-investigations` in [the codeowners file here](https://github.com/elastic/kibana/blob/main/.github/CODEOWNERS#L2622)?

agusruidiazgd committed:
fix tests + updated codeowners file

agusruidiazgd committed:
Merge branch 'main' into feat/attacks-alerts-flyout-table-fields-formatter

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/367029)
* Commit: 592cb8c94647121eb35bc3aac0b4eba87310c9e3

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/367029#019ad93e-ea8a-4f21-b982-c015c711400e)

### History
* :broken_heart: [Build #366366](https://buildkite.com/elastic/kibana-pull-request/builds/366366) failed b93d32c2b13ddbfddddd4fe803b3050d122945f3

cc @agusruidiazgd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367029","number":367029,"commit":"592cb8c94647121eb35bc3aac0b4eba87310c9e3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366366","number":366366,"commit":"b93d32c2b13ddbfddddd4fe803b3050d122945f3"}],"number":367029}
buildkite-pr-comment-->

agusruidiazgd committed:
fix typo

PhilippeOberti reviewed (approved):
Desk tested and code LGTM. Are you planning on doing the `Show more item` in a separate PR? I feel like this one could have been a good candidate as it focuses on the formatting of the table cells

agusruidiazgd committed:
Adding 'show more' funtion to show multiple values on table cell

agusruidiazgd commented:
> Desk tested and code LGTM. Are you planning on doing the `Show more item` in a separate PR? I feel like this one could have been a good candidate as it focuses on the formatting of the table cells

Thanks for the feedback, @PhilippeOberti! 🙏
You’re right, this PR was a good place to add the Show more functionality (I forgot to include it)

To address this properly I went ahead and implemented it here. I extracted the existing logic from the [highlighted_fields_cell](https://github.com/NicholasPeretti/kibana/blob/d7d237aad4d3c71093fbffc2d25c3fbc22b64f27/x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/highlighted_fields_cell.tsx) component and created a reusable hook called `use_expandable_values`. This lets us apply the same behavior consistently across other table cells that render regular values - wdyt?

e40pud commented on x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/shared/hooks/use_expandable_values.tsx:
```diff
@@ -0,0 +1,65 @@
+    };
+  }, [values, displayValuesLimit]);
+
+  const hasMultipleValues = (values?.length ?? 0) > 1;
```
Does not look like we use this variable?

e40pud commented on x-pack/solutions/security/plugins/security_solution/public/flyout/attack_details/components/table_field_value_cell.tsx:
```diff
@@ -0,0 +1,139 @@
+              width: auto;
+            `}
+          >
+            <EuiButtonEmpty
```
nit: Looks like we define `Show less/Show more` button in three different places - at least in this PR, probably in some other places as well. Maybe we could reuse it if it is the same everywhere.

agusruidiazgd replied:
Thanks for your insights! I'll include this refactor on a follow up PR with the changes implemented on document_details and IOC Flyouts ✅

e40pud reviewed (approved):
LGTM, thanks for the changes 🚀

PhilippeOberti commented on x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/shared/hooks/use_expandable_values.tsx:
```diff
@@ -0,0 +1,65 @@
+const EMPTY_ARRAY: string[] = [];
+
+interface UseExpandableValuesOptions {
+  values: string[] | null | undefined;
```
Let's have short description for those 2 props

PhilippeOberti reviewed (changes requested):
Now I think the PR does a bit too much! There is some (low) risk in making changes to the `document_details`, and the scope of the PR is now bigger.

I would consider undoing the changes to the `document_details` `highlighted_fields_cell` and `table_field_value` files. I think the `use_expandable_values` hook stil makes sense and could still live in the `document_details` folder, as we will use it in the follow up PR.
That wat this PR stays scoped to just `attack_details`, then the follow up PR can apply the changes to the `document_details` code as well as the `ios_details` code (and others if needed).

Regarding the button component Zhenia was commenting about, I think it's fine to keep it as is for now in this PR. But the follow up PR would extract that code in a reusable component and use it everywhere.

Let me know what you think! :)

agusruidiazgd committed:
remove changes from highlighted_fields_cell and table_field_value

agusruidiazgd committed:
apply feedback

agusruidiazgd committed:
Merge branch 'main' into feat/attacks-alerts-flyout-table-fields-formatter

PhilippeOberti reviewed (approved):
Desk tested and code LGTM! This looks awesome with the formatting and show more functionality!!!

agusruidiazgd committed:
Merge branch 'main' into feat/attacks-alerts-flyout-table-fields-formatter

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368836)
* Commit: c8be98d48278b1a623536dfd7956496432205ef1
* [Cloud Deployment](https://kibana-pr-244514-a0d72c.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #368455](https://buildkite.com/elastic/kibana-pull-request/builds/368455) failed c7381fbeb86bab6898f81f073bbdf84eaeb297fc
* :green_heart: [Build #367538](https://buildkite.com/elastic/kibana-pull-request/builds/367538) succeeded 4d1990b638d576e1217477d4614632637948ad9f
* :yellow_heart: [Build #367075](https://buildkite.com/elastic/kibana-pull-request/builds/367075) was flaky e8df50b393b6665a6dd19c5a17f4cfe0aa84fcee
* :broken_heart: [Build #367029](https://buildkite.com/elastic/kibana-pull-request/builds/367029) failed 592cb8c94647121eb35bc3aac0b4eba87310c9e3
* :broken_heart: [Build #366366](https://buildkite.com/elastic/kibana-pull-request/builds/366366) failed b93d32c2b13ddbfddddd4fe803b3050d122945f3

cc @agusruidiazgd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368836","number":368836,"commit":"c8be98d48278b1a623536dfd7956496432205ef1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368455","number":368455,"commit":"c7381fbeb86bab6898f81f073bbdf84eaeb297fc"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367538","number":367538,"commit":"4d1990b638d576e1217477d4614632637948ad9f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367075","number":367075,"commit":"e8df50b393b6665a6dd19c5a17f4cfe0aa84fcee"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367029","number":367029,"commit":"592cb8c94647121eb35bc3aac0b4eba87310c9e3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366366","number":366366,"commit":"b93d32c2b13ddbfddddd4fe803b3050d122945f3"}],"number":368836}
buildkite-pr-comment-->
